File: src/main/java/fr/khalypso/statesearchlib/model/State.java
package fr.khalypso.statesearchlib.model;

import java.util.List;


/**
 * State model class.
 * <p>
 * A state class
 * </p>
 */
public interface State {

	/**
	 * Is the state a terminal one.
	 */
	boolean isTerminal();

	/**
	 * {@link Operator} list that can be applied on this state.
	 */
	List<Operator> operators();

}

File: library/src/main/kotlin/fr/khalypso/statesearchlib/model/State.kt
package fr.khalypso.statesearchlib.model

/**
 * State model class.
 *
 *
 * A state class
 *
 */
interface State {
    /**
     * Is the state a terminal one.
     */
    val isTerminal: Boolean

    /**
     * [Operator] list that can be applied on this state.
     */
    val operators: List<Operator>
}
